deployers/deploymentreader.go

```diff
@@ -121,25 +126,32 @@ func (reader *DeploymentReader) bindPackageInputsAndAnnotations() {
 			serviceDeployPack.Package.Parameters = keyValArr
 		}
 
-		keyValArr = make(whisk.KeyValueArr, 0)
-
-		if len(pack.Annotations) > 0 {
+		if len(pack.Annotations) >0 {
+			// iterate over each annotation from deployment file
 			for name, input := range pack.Annotations {
-				var keyVal whisk.KeyValue
-
-				keyVal.Key = name
-				keyVal.Value = utils.GetEnvVar(input)
-
-				keyValArr = append(keyValArr, keyVal)
+				// check if annotation key in deployment file exists in manifest file
+				// setting a bool flag to false assuming key does not exist in manifest
+				keyExistsInManifest := false
+				// iterate over each annotation from manifest file
+				for i, a := range serviceDeployPack.Package.Annotations {
+					if name == a.Key {
+						// annotation key is found in manifest
+						keyExistsInManifest = true
+						// overwrite annotation in manifest file with deployment file
+						serviceDeployPack.Package.Annotations[i].Value = input
+						break
+					}
+				}
+				if !keyExistsInManifest {
+					return utils.NewYAMLFormatError("Annotation key \""+name+"\" does not exist in manifest file but specified in deployment file.")
+				}
 			}
-
-			serviceDeployPack.Package.Annotations = keyValArr
```

Comment:
Is the array always allocated now?

Reply:
Yes if one or more annotations are specified ...